src/jit/ee_il_dll.cpp

```diff
@@ -73,13 +73,19 @@ void __stdcall jitStartup(ICorJitHost* jitHost)
     if (jitstdout == nullptr)
     {
         int jitstdoutFd = _dup(_fileno(procstdout()));
```

Comment:
> Should the check be applied to _fileno as well since it can invoke the invalid_param_handler

AFAICT the problem does not appear to be the invalid param handler: if it's what is causing the crash that powershell is seeing, then the crash shouldn't be happening at the call to`_setmode`, but at either the call to `_fileno` or more likely `_dup` (the former would only invoke the error is `stdout` was `NULL`, which does not appear to be the case in the Powershell repro). That said, it is not a bad idea to check the output of both functions. As per MSDN, the check on `_fileno` needs to check for `-1` and `-2`. For example:

```
int stdoutFd = _fileno(procstdout());
if (stdoutFd != -1 && stdoutFd != -2)
{
    int jitstdoutFd = _dup(stdoutFd);
    if (jitstdoutFd != -1)
    {
        ...
    }
}

if (jitstdout == nullptr)
{
    jitstdout = procstdout();
}
```